cpp/include/cuvs/neighbors/ivf_pq_helpers.cuh

```diff
@@ -0,0 +1,792 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
```

Comment:
I won't exhaustively go through each of these files, so this comment will apply to all remaining. The `include/` directory should contain 1) no implementations, and 2) only definitions for pre-compiled source files in cpp/.